src/main/java/core/basesyntax/service/ShopServiceImpl.java

```diff
@@ -0,0 +1,24 @@
+package core.basesyntax.service;
+
+import core.basesyntax.model.FruitTransaction;
+import java.util.List;
+
+public class ShopServiceImpl implements ShopService {
+    private ActionStrategy strategy;
+
+    public ShopServiceImpl(ActionStrategy strategy) {
+        this.strategy = new ActionStrategyImpl(strategy);
+    }
```

Comment:
This doesn't make much sense, did you try to implement the constructor dependency injection pattern?
You probably meant to do `this.strategy = strategy;` You will create actual impl and pass it to ShopServiceImpl constructor in your main class, not here. You're passing actionstrategy to constructor of actionStrategyImpl, this is weird. Dependency injection is about decoupling the creation of dependencies from the class that uses them.